File: Dec2020/15Dec/ProfitsByCars.cpp
/*
Problem Name: Profits by cars
Problem Link: https://www.hackerearth.com/practice/algorithms/greedy/basics-of-greedy-algorithms/practice-problems/algorithm/jetha-and-cars-2d835131/
*/

#include<bits/stdc++.h>
using namespace std;

#define test(t)     int t; cin>>t; while(t--)

int main(){
    ios::sync_with_stdio(0);    
    cin.tie(0);             
    test(t){
        int n;
        cin>>n;
        int P[n];
        for(int i=0;i<n;++i)
            cin>>P[i];
        
        sort(P,P+n);
        long sum=P[0], prev=P[0];
        for(int i=1;i<n;++i){
            if(P[i] != prev){
                sum += P[i];
                prev = P[i];
            }
        }
        cout<<sum<<"\n";
    }
    return 0;
}

File: Dec2020/15Dec/ProfitsByCars.java
/*
Problem Name: Profits by cars
Problem Link: https://www.hackerearth.com/practice/algorithms/greedy/basics-of-greedy-algorithms/practice-problems/algorithm/jetha-and-cars-2d835131/
*/
import java.io.*;
import java.util.*;
class TestClass {
    public static void main(String args[] ) throws Exception {
        Scanner in = new Scanner(System.in);
        PrintWriter out = new PrintWriter(System.out);
        int t = in.nextInt();
        for(;t>0;t--){
            int n = in.nextInt();
            int arr[] = new int[n];
            for(int i=0;i<n;i++)
                arr[i] = in.nextInt();
            Arrays.sort(arr);
            long sum = arr[0];
            for(int i=1;i<n;i++)
                if(arr[i] != arr[i-1])
                    sum += arr[i];
            out.println(sum);
        }
        out.flush();
    }
}
